src/etrans/cpu/CMakeLists.txt

```diff
@@ -47,11 +47,71 @@ endfunction(generate_backend_sources)
 
 set( BUILD_INTERFACE_INCLUDE_DIR ${CMAKE_BINARY_DIR}/include/ectrans )
 
+function(generate_etrans_dummy_sources)
+  set (options)
+  set (oneValueArgs BACKEND DESTINATION OUTPUT)
+  set (multiValueArgs)
+
+  cmake_parse_arguments(_PAR "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+  set(backend ${_PAR_BACKEND})
+  set(destination ${_PAR_DESTINATION})
+
+  file(MAKE_DIRECTORY ${destination}/external)
+
+  # Build list of interface files to generate dummy sources for
+  ecbuild_list_add_pattern( LIST include_files
+    GLOB etrans/*.h
+    SOURCE_DIR ${PROJECT_SOURCE_DIR}/src/etrans/include
+    QUIET
+  )
+
+  foreach( include_file ${include_files} )
+    cmake_path( GET include_file STEM file_stem )
+    set(dummy_source "${file_stem}.F90")
+
+    # First generate a dummy source file from the interface that just contains an abort
+    file( READ ${include_file} contents )
+    string( REPLACE "END INTERFACE" "" contents "${contents}" )
+    string( REPLACE "INTERFACE" "" contents "${contents}" )
+    string(
+      REPLACE "IMPLICIT NONE" "USE ABORT_TRANS_MOD, ONLY: ABORT_TRANS\nIMPLICIT NONE"
+      contents "${contents}"
+    )
+    string(
+      REPLACE "END SUBROUTINE"
+        "CALL ABORT_TRANS(\"etrans dummy library - you should not be here\")\nEND SUBROUTINE"
+      contents "${contents}"
+    )
+    file( WRITE ${CMAKE_CURRENT_BINARY_DIR}/generated/etrans_dummy/${dummy_source} "${contents}" )
+
+    # Then run the dummy source through the normal backend file generator
+    set(outfile "${destination}/${dummy_source}")
+    ecbuild_debug("Generate ${outfile}")
+    generate_file(BACKEND ${backend} INPUT ${CMAKE_CURRENT_BINARY_DIR}/generated/etrans_dummy/${dummy_source} OUTPUT ${outfile})
+    list(APPEND outfiles ${outfile})
+  endforeach( include_file )
+  set(${_PAR_OUTPUT} ${outfiles} PARENT_SCOPE)
+endfunction(generate_etrans_dummy_sources)
+
 foreach( prec dp sp )
   if( HAVE_${prec} )
 
     generate_backend_includes(BACKEND ${prec} TARGET ectrans_etrans_${prec}_includes DESTINATION ${BUILD_INTERFACE_INCLUDE_DIR} INCLUDE_DIRECTORY ${PROJECT_SOURCE_DIR}/src/etrans/include )
-    generate_backend_sources( BACKEND ${prec} OUTPUT ectrans_etrans_${prec}_src  DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/generated/ectrans_etrans_${prec})
+
+    if( NOT HAVE_ETRANS )
```

Comment:
I guess longer term we might want to consider making the dummy generation optional, too. But fine for this PR.

Reply:
I think we can do this indeed but then provided that ifs-source can link to ectrans without any symbols needed.